akka-http-core/src/main/scala/akka/http/impl/engine/rendering/RenderSupport.scala

```diff
@@ -79,13 +79,13 @@ private object RenderSupport {
     override def onPush(elem: ByteString, ctx: Context[ByteString]): SyncDirective = {
       sent += elem.length
       if (sent > length)
-        throw InvalidContentLengthException(s"HTTP message had declared Content-Length $length but entity data stream amounts to more bytes")
+        ctx fail InvalidContentLengthException(s"HTTP message had declared Content-Length $length but entity data stream amounts to more bytes")
```

Comment:
I don't know, this failure shouldn't happen in normal circumstances or what am I missing? I.e. why are there log outputs of this error in our tests? If they are expected in a test then they should be intercepted instead. If they are expected as in ("not a big error but hey") then log a warning.